lib/rules/array-callback-return.js

```diff
@@ -188,19 +193,36 @@ module.exports = {
          * @returns {void}
          */
         function checkLastSegment(node) {
-            if (funcInfo.shouldCheck &&
-                funcInfo.codePath.currentSegments.some(isReachable)
-            ) {
-                context.report({
-                    node,
-                    loc: getLocation(node, context.getSourceCode()).loc.start,
-                    messageId: funcInfo.hasReturn
-                        ? "expectedAtEnd"
-                        : "expectedInside",
-                    data: {
-                        name: astUtils.getFunctionNameWithKind(funcInfo.node)
+            const method = astUtils.getStaticPropertyName(funcInfo.node.parent.callee);
+
+            if (funcInfo.shouldCheck) {
+                if (method === "forEach") {
+                    if (options.checkForEach &&
+                        !funcInfo.codePath.currentSegments.some(isReachable)
+                    ) {
+                        context.report({
```

Comment:
Should this be a correct or incorrect code:

```js
/*eslint array-callback-return: ["error", { checkForEach: true }]*/

myArray.forEach(function(item) {
    if (item < 0) {
        return false;
    }
    handleItem(item);
});
```

If this option should report any `return value`, I'm not sure that the current approach is right. It seems that this condition checks only if there is at least one path that doesn't return/throw.

Reply:
It should report any return value.